File: groceryapp-frontend/src/services/storeService.js
import api from "./api";

const BASE_URL = "/stores";

export const getStores = () => api.get(BASE_URL);
export const getStoreById = (id) => api.get(`${BASE_URL}/${id}`);
export const addStore = (data) => api.post(BASE_URL, data);
export const updateStore = (id, data) => api.put(`${BASE_URL}/${id}`, data);
export const deleteStore = (id) => api.delete(`${BASE_URL}/${id}`);

File: src/main/java/com/grocery/groceryapp/service/StoreService.java
package com.grocery.groceryapp.service;

import com.grocery.groceryapp.entity.Store;
import com.grocery.groceryapp.repository.StoreRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class StoreService {

    @Autowired
    private StoreRepository storeRepository;

    // Create Store
    public Store saveStore(Store store) {
        return storeRepository.save(store);
    }

    // Get all Stores
    public List<Store> getAllStores() {
        return storeRepository.findAll();
    }

    // Get Store by ID
    public Store getStoreById(Long id) {
        return storeRepository.findById(id).orElse(null);
    }

    // Update Store
    public Store updateStore(Long id, Store store) {
        Store existing = storeRepository.findById(id).orElse(null);

        if (existing != null) {
            existing.setStoreName(store.getStoreName());
            existing.setStoreLocation(store.getStoreLocation());
            return storeRepository.save(existing);
        }

        return null;
    }

    // Delete Store
    public void deleteStore(Long id) {
        storeRepository.deleteById(id);
    }

}
